### PR TITLE
Fix array return type in python

### PR DESCRIPTION
**Pull request by IgorBaratta**

### CODE REVIEW

File: python/dolfinx/wrappers/la.cpp [hunk 1]
```diff
@@ -96,8 +96,17 @@ void la(py::module& m)
...
              std::shared_ptr<dolfinx::la::Vector<PetscScalar>>>(m, "Vector")
-      .def_property_readonly("array",
-                             &dolfinx::la::Vector<PetscScalar>::mutable_array);
+      .def_property_readonly(
+          "array",
+          [](dolfinx::la::Vector<PetscScalar>& self) {
+            std::vector<PetscScalar>& array = self.mutable_array();
+            return py::array(array.size(), array.data(), py::cast(self));
+          })
+      .def_property_readonly(
```
Comment (garth-wells) [inline python/dolfinx/wrappers/la.cpp:105]: Why are there two near-identical functions?

Comment (IgorBaratta) [reply python/dolfinx/wrappers/la.cpp:105]: One for `dolfinx::la::Vector<PetscScalar>`  and the other for `std::shared_ptr<dolfinx::la::Vector<PetscScalar>>`.
There is possibly a smarter way to do this.. but I'm not aware of one alternative.

Comment (garth-wells) [reply python/dolfinx/wrappers/la.cpp:105]: I see that, but why is it needed?

Comment (IgorBaratta) [reply python/dolfinx/wrappers/la.cpp]: It's not actually needed.
It was just my misunderstanding of how smart pointers work in pybind.
https://pybind11.readthedocs.io/en/stable/advanced/smart_ptrs.html#std-shared-ptr

Just removed the first function.

File: python/dolfinx/wrappers/la.cpp [hunk 2]
```diff
@@ -96,8 +96,11 @@ void la(py::module& m)
   // dolfinx::la::Vector
   py::class_<dolfinx::la::Vector<PetscScalar>,
              std::shared_ptr<dolfinx::la::Vector<PetscScalar>>>(m, "Vector")
-      .def_property_readonly("array",
-                             &dolfinx::la::Vector<PetscScalar>::mutable_array);
+      .def_property_readonly(
```
Comment (garth-wells) [inline python/dolfinx/wrappers/la.cpp:99]: Keep the reference version - it's simpler and consistent with what we do elsewhere.

**Review by garth-wells — APPROVED** [review]

**Pull request merged**